data/web/index.html

```diff
@@ -264,13 +264,35 @@ <h1 id="showSelectedGroup"></h1>
             }
         }
 
-        $(document).ready(function() {
-            updateTimer();
-            createGroups();
-            selectGroup(0);
-            createResultTable();
-            updateResultTable();
+        function getGroups() {
+            global.wsClient.getGroups().then(function (rsp) {
+                global.numberOfGroups = rsp.groups;
+                console.log(global.numberOfGroups + " Groups are configured.");
+                console.info("Retrieved Groups.");
+                updateTimer();
+                createGroups();
+                selectGroup(0);
+                createResultTable();
+                updateResultTable();

```

Comment:
I am having trouble chaining these functions, as I am not sure how would a Promise work here.  my getGroups() function is not a promise in index.html, however is calls a promise from ws.js. How can I wait for this "indirect" promise to be done before continuing with the next function?

Reply:
This would be the ideal way to go:

```
$(document).ready(function () {
            getGroups();
            updateTimer();
            createGroups();
            selectGroup(0);
            createResultTable();
            getSavedTable();
            updateResultTable();
```

Reply:
```javascript
getGroups().then(function() {
    /* Do something here ... */
    return Promise.resolve();
}.then(function() {
    /* In case you have a function which return a promise. */
    return functionWithPromise();
}.then(function() {
    /* ... */
}
```

Reply:
I get an error: `Cannot read property 'then' of undefined` on the client side.  Is it because getGroups() is not a promise but a function that calls a promise?

Reply:
You must return the promise, which you use in getGroups() or return a new promise.